Framework/MDAlgorithms/src/IntegratePeaksMDHKL.cpp

```diff
@@ -0,0 +1,319 @@
+                  }
+                }
+              }
+            }
+            if (neighborPoints >= neighborPts) {
+              peakPoints = peakPoints + 1;
+              peakSum = peakSum + F[iHKL];
+              errSqSum = errSqSum + SqError[iHKL];
+            }
+          }
+        } else {
+          double minR =
+              sqrt(std::pow(float(Hindex) / float(gridPts[0]) - 0.5, 2) +
+                   std::pow(float(Kindex) / float(gridPts[1]) - 0.5, 2) +
+                   std::pow(float(Lindex) / float(gridPts[0]) - 0.5, 2));
+          if (minR < 0.05) {
+            intensity = 0.0;
+            errorSquared = 0.0;
+            return;
+          }
+        }
+      }
+    }
+  }
+  double ratio = float(peakPoints) / float(measuredPoints - peakPoints);
+  intensity = peakSum - ratio * (measuredSum - peakSum);
+  errorSquared = errSqSum + ratio * (measuredErrSqSum - errSqSum);
+  return;
+}
+
+/**
+ * Runs the BinMD algorithm on the input to provide the output workspace
+ * All slicing algorithm properties are passed along
+ * @return MDHistoWorkspace as a result of the binning
+ */
+MDHistoWorkspace_sptr IntegratePeaksMDHKL::binEvent(int h, int k, int l,
+                                                    double box, int gridPts,
+                                                    IMDWorkspace_sptr ws) {
+  IAlgorithm_sptr binMD = createChildAlgorithm("BinMD", 0.0, 0.3);
+  binMD->setProperty("InputWorkspace", ws);
+  binMD->setProperty("AlignedDim0",
+                     "[H,0,0]," + boost::lexical_cast<std::string>(h - box) +
+                         "," + boost::lexical_cast<std::string>(h + box) + "," +
+                         boost::lexical_cast<std::string>(gridPts));
+  binMD->setProperty("AlignedDim1",
+                     "[0,K,0]," + boost::lexical_cast<std::string>(k - box) +
+                         "," + boost::lexical_cast<std::string>(k + box) + "," +
+                         boost::lexical_cast<std::string>(gridPts));
+  binMD->setProperty("AlignedDim2",
+                     "[0,0,L]," + boost::lexical_cast<std::string>(l - box) +
+                         "," + boost::lexical_cast<std::string>(l + box) + "," +
+                         boost::lexical_cast<std::string>(gridPts));
+  binMD->setPropertyValue("AxisAligned", "1");
+  binMD->setPropertyValue("OutputWorkspace", "out");
+  binMD->executeAsChildAlg();
+  Workspace_sptr outputWS = binMD->getProperty("OutputWorkspace");
+  return boost::dynamic_pointer_cast<MDHistoWorkspace>(outputWS);
+}
+
+/**
+ * Runs the BinMD algorithm on the input to provide the output workspace
+ * All slicing algorithm properties are passed along
+ * @return MDHistoWorkspace as a result of the binning
+ */
+MDHistoWorkspace_sptr IntegratePeaksMDHKL::cropHisto(int h, int k, int l,
+                                                     double box,
+                                                     IMDWorkspace_sptr ws) {
```

Comment:
Since the `IPropertyManager` takes `ws` by const reference, should we also pass it here by const reference?